core/src/main/java/com/linecorp/armeria/client/HttpClientDelegate.java

```diff
@@ -202,11 +202,13 @@ private void doExecute(PooledChannel pooledChannel, ClientRequestContext ctx,
             final HttpSession session = HttpSession.get(channel);
             res.init(session.inboundTrafficController());
             final SessionProtocol sessionProtocol = session.protocol();
+
+            // Should never reach here.
             if (sessionProtocol == null) {
                 needsRelease = false;
                 try {
                     // TODO(minwoox): Make a test that handles this case
-                    final UnprocessedRequestException cause = UnprocessedRequestException.get();
+                    final NullPointerException cause = new NullPointerException("sessionProtocol");
```

Comment:
I changed this into NullPointerException because it is abnormal behavior and not a retriable case.